xload/options.go

```diff
@@ -0,0 +1,44 @@
+package xload
+
+const defaultKey = "config"
+
+// Option configures the xload behaviour.
+type Option interface {
+	apply(*options)
+}
+
+// OptionFunc allows using a function as an Option.
+type OptionFunc func(*options)
```

Comment:
```diff
--- type OptionFunc func(*options)
+++ type optionFunc func(*options)
```

Since neither `options` nor its fields are public, exposing this doesn't add value.